src/osd/OSDMap.h

```diff
@@ -1720,7 +1720,8 @@ bool try_drop_remap_underfull(
     read_balance_info_t &p_rb_info) const;
 
   float rbi_round(float f) const {
-    return (f > 0.0) ? floor(f * 100 + 0.5) / 100 : ceil(f * 100 - 0.5) / 100;
+    // Round to two decimal places away from zero to ensure the result does not become 0
+    return (f > 0.0) ? ceil(f * 100 + 0.5) / 100 : floor(f * 100 - 0.5) / 100;
```

Comment:
IIUC originally the function result for f==0.0 was equal to 0. With this patch it becomes -1/100. Is this expected/safe?